### DISCUSSION TITLE
FastAPI worker timing out

### DISCUSSION THREAD

**Discussion by ArijitSinghEDA** [fastapi/fastapi#10620] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel
import asyncio
import uvicorn

app = FastAPI()

class Item(BaseModel):
    sql: str

@app.post("/out")
async def func(item: Item):
    from functionality import main
    asyncio.ensure_future(main(query=item.sql))
    return 1

if __name__=="__main__":
    uvicorn.run(app, host="127.0.0.1", port=8888)
```

### Description

I am running the code using gunicorn inside my virtual environment with the command 
`python -m gunicorn main:app --bind 127.0.0.1:8888 --workers 4 --worker-class uvicorn.workers.UvicornWorker --timeout 86400 --graceful-timeout 86400`

Inside my `main` function, I am reading an sql query and converting it into a pandas dataframe. When the query has 5M rows, there's no issue, but when the number of rows increase to 10M, the workers are sent a `SIGKILL` signal, even after setting both `timeout` and `graceful-timeout`. Is there any internal timeout set by `fastAPI`? I searched all over the internet but couldn't find anything suggesting that fastAPI can behave like this.

### Operating System

Linux

### Operating System Details

I am using `wsl` on my Windows system.

### FastAPI Version

0.104.1

### Pydantic Version

2.4.2

### Python Version

3.10.12

### Additional Context

I am running this from inside a virtual environment.

**Comment by theobouwman** [comment]
Maybe you dont have enough memory?

**Reply by jgould22** [reply]
Mentioned in the docs here https://docs.gunicorn.org/en/stable/faq.html?highlight=SIGKILL#why-are-workers-silently-killed

**Reply by ArijitSinghEDA** [reply]
I made the same dataset and loaded on my local system without any problem, and a pandas dataframe also gets loaded into the memory. So does it mean fastAPI/gunicorn is taking excessive memory?

**Reply by ArijitSinghEDA** [reply]
And I checked my memory consumption. It maximum to 94%, but the worker restarts after consumption was 86% (other necessary processes are also running in the background, so this memory consumption percentage is for the whole system)

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
